lldb/source/Symbol/SwiftASTContext.cpp

```diff
@@ -1464,22 +1464,27 @@ void SwiftASTContext::AddExtraClangArgs(std::vector<std::string> ExtraArgs) {
         clang_argument == "-working-directory")
       continue;
 
+    auto clear_arg = llvm::make_scope_exit([&] { clang_argument.clear(); });
```

Comment:
Can we detect -Werror earlier, like:

```
for (... argString : ExtraArgs) {
  StringRef arg(argString);
  if (arg.startswith("-Werror")) continue;
```

I expect we'll want to make similar changes for other kinds of options that don't affect debugging.

Reply:
See my other comment below.

Reply:
Why does the multi-arg macro handling make it necessary to check for -Werror late?